### ISSUE TITLE
Uncaught ReferenceError: global is not defined

### ISSUE THREAD

**Issue by linfanxxxx** [remcohaszing/monaco-yaml#73]
need some help.

I use `monaco-editor`, `monaco-yaml`, `monaco-editor-webpack-plugin` in Angular.

But I get an error such like this

![image](https://user-images.githubusercontent.com/34636576/128887336-6b5bcbe2-ed13-4769-b2e9-a71762a50007.png)

- how I use 

webpack.config.js
```
const MonacoEditorWebpackPlugin = require("monaco-editor-webpack-plugin");
const webpack = require("webpack");

module.exports = (config, options) => {
  config.plugins.push(
    new MonacoEditorWebpackPlugin({
      languages: ['yaml'],
      globalAPI: true,
      customLanguages: [
        {
          label: "yaml",
          entry: "../../monaco-yaml/lib/esm/monaco.contribution",
          worker: {
            id: "vs/language/yaml/yamlWorker",
            entry: "../../monaco-yaml/lib/esm/yaml.worker.js",
          },
        },
      ],
    })
  );
  return config;
};
```

app.component.ts
```
import { AfterViewInit, Component, OnInit } from '@angular/core';

import * as monaco from 'monaco-editor/esm/vs/editor/editor.api.js';
@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent implements OnInit, AfterViewInit {
  title = 'my-app'
  constructor(){

  }
  ngAfterViewInit(): void {
    monaco.editor.create(document.getElementById('testEditor'), {
      value: `
object: 
  test: 123
      `,
      language: 'yaml',
      theme: "vs-dark",
    });

    console.log('monaco2', monaco)
    setTimeout(()=>{
      console.log('monaco', monaco)
    },1000)
  }
  ngOnInit(): void {
  }
}
```

and I make a demo for debug

here is repo : https://github.com/linfanxxxx/monaco-debug/tree/master/src

run it with below:

> npm install
> npm run start

and you will see the error

**Comment by remcohaszing** [comment]
Having a link to the repository is really helpful! :smile: 

Does it work if you set `globalAPI: false` in the `MonacoEditorWebpackPlugin` options?

Personally I recommend to not use `monaco-editor-webpack-plugin` at all. It’s just not necessary. Since you’re using Webpack 4 (based on your `package-lock.json`), I recommend having a look at https://github.com/remcohaszing/monaco-yaml/tree/v3.0.0/examples/webpack-worker-loader.

I think the following section should just make it work:

https://github.com/remcohaszing/monaco-yaml/blob/090302534ae205d3a7163375bf211be5e8dd5eda/examples/webpack-worker-loader/src/index.js#L3-L21

Note that you **must** use Webpack loader syntax because of https://github.com/microsoft/monaco-editor/issues/2592

And maybe you need this to make icons work. I’m not familiar with Angular build tooling.

https://github.com/remcohaszing/monaco-yaml/blob/090302534ae205d3a7163375bf211be5e8dd5eda/examples/webpack-worker-loader/webpack.config.js#L10-L13

**Comment by remcohaszing** [comment]
I recently found out [unikubehq/frontend](https://github.com/unikubehq/frontend) ([see here](https://github.com/unikubehq/frontend/search?q=monaco-yaml&type=code)) also uses `monaco-yaml` with `monaco-webpack-plugin`. Does this help?

**Comment by remcohaszing** [comment]
I’m going to close this issue, because the same issue was resolved in #98
